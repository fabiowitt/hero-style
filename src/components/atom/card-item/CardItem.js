import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import css from './CardItem.module.scss';

import EmptyCardItem from '../empty-card-item/EmptyCardItem';

const CardItem = ({ format, label, leftIcon, value, ...otherProps }) => {
  const renderContent = (_label, _value, _leftIcon) => (
    <>
      {_leftIcon || null}
      <div className={css['c-card-item__content']}>
        <span>{_label}</span>
        <span className={css.truncate}>
          {_value instanceof Date ? moment(_value).format(format) : _value}
        </span>
      </div>
    </>
  );

  return (
    <EmptyCardItem {...otherProps}>
      {renderContent(label, value, leftIcon)}
    </EmptyCardItem>
  );
};

CardItem.propTypes = {
  ellipsis: PropTypes.bool,
  format: PropTypes.string,
  gray: PropTypes.bool,
  label: PropTypes.string,
  lbPrimary: PropTypes.bool,
  lbWhite: PropTypes.bool,
  lbGray: PropTypes.bool,
  leftIcon: PropTypes.instanceOf(Object),
  primary: PropTypes.bool,
  white: PropTypes.bool,
  small: PropTypes.bool,
  svgPrimary: PropTypes.bool,
  svgWhite: PropTypes.bool,
  valPrimary: PropTypes.bool,
  valWhite: PropTypes.bool,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.instanceOf(Date),
    PropTypes.number
  ])
};

CardItem.defaultProps = {
  ellipsis: false,
  format: 'DD MMM YYYY [Ã s] HH:mm',
  gray: false,
  label: null,
  lbPrimary: false,
  lbWhite: false,
  lbGray: false,
  leftIcon: null,
  primary: false,
  white: false,
  small: false,
  svgPrimary: false,
  svgWhite: false,
  valPrimary: false,
  valWhite: false,
  value: null
};

CardItem.displayName = 'CardItem';

export default CardItem;
