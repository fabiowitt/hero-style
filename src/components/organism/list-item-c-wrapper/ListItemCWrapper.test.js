import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ListItemCWrapper from './ListItemCWrapper';
import CardItem from '../../atom/card-item/CardItem';
import ListItemContent from '../../molecule/list-item-content/ListItemContent';

describe('ListItemCWrapper', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(
      <ListItemCWrapper>
        <ListItemContent>
          <CardItem value="Value" label="Label" />
          <CardItem value="Value" label="Label" />
          <CardItem value="Value" label="Label" />
        </ListItemContent>
      </ListItemCWrapper>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
