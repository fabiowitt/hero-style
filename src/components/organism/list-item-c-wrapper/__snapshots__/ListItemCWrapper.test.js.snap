// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemCWrapper should match with the snapshot 1`] = `
<div
  className="c-wrapper"
>
  <ListItemContent
    border={false}
    isAction={false}
  >
    <CardItem
      col={null}
      ellipsis={false}
      format="DD MMM YYYY [às] HH:mm"
      label="Label"
      primary={false}
      value="Value"
    />
    <CardItem
      col={null}
      ellipsis={false}
      format="DD MMM YYYY [às] HH:mm"
      label="Label"
      primary={false}
      value="Value"
    />
    <CardItem
      col={null}
      ellipsis={false}
      format="DD MMM YYYY [às] HH:mm"
      label="Label"
      primary={false}
      value="Value"
    />
  </ListItemContent>
</div>
`;
